uiowa/uiowa
# [Registrar] Academic Calendar

Resolves #7788 
Resolves #7853

# How to test
```
ddev blt frontend && ddev blt ds --site=registrar.uiowa.edu && ddev drush @registrar.local uli /node/321/layout
```
1. Place "Academic calendar" block and select "Current session plus next 5 sessions"
2. Test whether this meets the non-calendar portions of the proposed solution on #7788 and #7853
3. Page back and forth and test how the calendar loads events
4. Code review
5. Comment out the [printDate flag](https://github.com/uiowa/uiowa/pull/7781/files#diff-2a4f5378c573b7d600040665593e8fb44c96a0899cf8141283d047349c4d9186R136) and the [reviewed filter](https://github.com/uiowa/uiowa/pull/7781/files#diff-2a4f5378c573b7d600040665593e8fb44c96a0899cf8141283d047349c4d9186R139-R141) and set a breakpoint on the number of $events returned at the end of fetchAndProcessCalendarData to see that those filters are working.
6. Note that the frontend is now being served by a card render array instead of the JS.
